client/signup/steps/wp-for-teams-site/style.scss

```diff
@@ -20,3 +20,10 @@
 .wp-for-teams-site__validation-site-title.validation-fieldset {
 	margin-bottom: 20px;
 }
+
+.signup.is-wp-for-teams .form-label {
+	&:hover {
+		cursor: pointer;
```

Comment:
I would remove this — the cursor shouldn't change when hovering the label, it's only an aid to make the field easier to click.